src/components/body-controller.js

```diff
@@ -0,0 +1,48 @@
+AFRAME.registerComponent("body-controller", {
+  schema: {
+    camera: { type: "selector", default: "[camera]" },
+    rotationSpeed: { default: 0.07 },
+    eyeNeckOffset: { type: "vec3" },
+    neckHeight: { type: "number" }
+  },
+
+  init() {
+    this.targetAngle = new THREE.Quaternion();
+    this.cameraPositionMatrix = new THREE.Matrix4();
+    const offset = this.data.eyeNeckOffset;
+    this.eyeNeckTransformMatrix = new THREE.Matrix4().makeTranslation(
+      offset.x,
+      offset.y,
+      offset.z
+    );
+    this.neckTransformMatrix = new THREE.Matrix4().makeTranslation(
+      0,
+      this.data.neckHeight,
+      0
+    );
+    this.bodyPositionMatrix = new THREE.Matrix4();
+    this.bodyPositionVector = new THREE.Vector3();
+  },
+
+  tick(time, dt) {
+    const object3D = this.el.object3D;
+    const cameraObject3D = this.data.camera.object3D;
+
+    // Set Rotation
+    const cameraQuat = cameraObject3D.quaternion;
+    this.targetAngle.set(0, cameraQuat.y, 0, cameraQuat.w);
```

Comment:
I have not seen someone set the fields of quaternion manually like this to get a desired rotation. How does this work?